tests/interpreter_tests.rs

```diff
@@ -0,0 +1,394 @@
+                    .map(|_| {
+                        let mut val = rng.next_u64();
+                        if *bw != 64 {
+                            val = val & ((1 << bw) - 1);
+                        }
+                        val
+                    })
+                    .collect();
+                env.insert(name, vals.clone());
+                vals
+            })
+            .collect();
+
+        for t in 0..num_clock_cycles {
+            for input in input_values.iter() {
+                sim_proc_stdin
+                    .write_all(format!("{:X}\n", input[t]).as_bytes())
+                    .unwrap();
+            }
+            let result = interpret(&serialized, &root_node.eclass, t, &env).unwrap();
+            interpreter_results.push(result);
+        }
+    }
+
+    let output = sim_proc.wait_with_output().unwrap();
+
+    let output_str = String::from_utf8(output.stdout).unwrap();
+    let verilator_output_values: Vec<u64> = output_str
+        .lines()
+        // filter all lines that don't start with "output: "
+        .filter(|line| line.len() > 0 && line.starts_with("output: "))
+        .map(|line| line.trim_start_matches("output: ").parse().unwrap())
+        .collect();
+
+    for (InterpreterResult::Bitvector(val, _), verilator_result) in interpreter_results
+        .iter()
+        .zip(verilator_output_values.iter())
+    {
+        assert_eq!(val, verilator_result);
+    }
+
+    let test_output_path = test_output_dir.join("test_output.txt");
+    let test_error_path = test_output_dir.join("test_error.txt");
+
+    std::fs::write(&test_output_path, output_str).unwrap();
+    std::fs::write(&test_error_path, output.stderr).unwrap();
+
+    println!("logged output to: {}", test_output_path.to_str().unwrap());
+}
+
+macro_rules! interpreter_test {
+    ($test_name:ident, $expected:expr, $verilog_path:literal, $module_name:literal, $time:literal, $env:expr, $out: literal) => {
+        #[test]
+        fn $test_name() {
+            let (serialized, root_node) = prep_interpreter(
+                PathBuf::from($verilog_path),
+                std::env::temp_dir(),
+                $module_name,
+                $out,
+            );
+
+            assert_eq!(
+                $expected,
+                interpret(&serialized, &root_node.eclass, $time, $env).unwrap()
+            );
+        }
+    };
+}
+
+interpreter_test!(
+    test_alu_0,
+    InterpreterResult::Bitvector(0b01010101, 8),
+    "tests/interpreter_tests/verilog/ALU.sv",
+    "ALU",
+    0,
+    &[
+        ("a", vec![0b01010101]),
+        ("b", vec![0b11111111]),
+        ("op", vec![1])
+    ]
+    .into(),
+    "out"
+);
+
+interpreter_test!(
+    test_alu_1,
+    InterpreterResult::Bitvector(0b11111111, 8),
+    "tests/interpreter_tests/verilog/ALU.sv",
+    "ALU",
+    0,
+    &[
+        ("a", vec![0b01010101]),
+        ("b", vec![0b11111111]),
+        ("op", vec![0])
+    ]
+    .into(),
+    "out"
+);
```

Comment:
Can you add a test that goes beyond t=0?

Reply:
Done!